app.js

```diff
@@ -25,10 +28,19 @@ app.get('/impersonations/current', async function (req, res, next) {
   const data = {
     type: 'impersonations',
     id: sessionId,
+    attributes: {
+      'original-session-roles': originalSessionRoles,
```

Comment:
This  might not be needed if we store the original roles and group in the session data of the frontend. It would have the benefit that the service stays more generic.